app/src/main/java/org/sagebionetworks/research/mpower/tracking/recycler_view/SymptomsLoggingItemViewHolder.java

```diff
@@ -0,0 +1,250 @@
+import org.sagebionetworks.research.mpower.tracking.view_model.configs.SimpleTrackingItemConfig;
+import org.sagebionetworks.research.mpower.tracking.view_model.logs.SymptomLog;
+import org.sagebionetworks.research.mpower.tracking.view_model.TrackingTaskViewModel;
+import org.sagebionetworks.research.mpower.tracking.widget.SymptomsLoggingUIFormItemWidget;
+import org.threeten.bp.Instant;
+import org.threeten.bp.ZoneId;
+import org.threeten.bp.ZonedDateTime;
+import org.threeten.bp.format.DateTimeFormatter;
+import org.threeten.bp.zone.ZoneRulesException;
+
+import java.util.List;
+
+/**
+ * View Holder for the Logging Items in the Symptoms task.
+ */
+public class SymptomsLoggingItemViewHolder extends RecyclerView.ViewHolder {
+    private static final float FULL_ALPHA = 1.0f;
+    private static final float FADED_ALPHA = .35f;
+
+    private TrackingTaskViewModel<SimpleTrackingItemConfig, SymptomLog> viewModel;
+    private SymptomsLoggingUIFormItemWidget widget;
+    private SymptomLoggingFragment symptomLoggingFragment;
+
+    private RadioButton previousSelectedSeverityButton;
+
+    public SymptomsLoggingItemViewHolder(final SymptomsLoggingUIFormItemWidget itemView, final
+    TrackingTaskViewModel<SimpleTrackingItemConfig, SymptomLog> viewModel, final SymptomLoggingFragment symptomLoggingFragment) {
+        super(itemView);
+        this.widget = itemView;
+        this.viewModel = viewModel;
+        this.symptomLoggingFragment = symptomLoggingFragment;
+    }
+
+    public void setContent(@NonNull SimpleTrackingItemConfig config) {
+        TrackingItem trackingItem = config.getTrackingItem();
+        this.setLogObservers(trackingItem);
+
+        // Setup the title and detail labels
+        this.widget.getTitle().setText(trackingItem.getIdentifier());
+        String detail = trackingItem.getDetail();
+        if (detail != null) {
+            this.widget.getDetail().setVisibility(View.VISIBLE);
+            this.widget.getDetail().setText(detail);
+        } else {
+            this.widget.getDetail().setVisibility(View.GONE);
+        }
+
+        // Setup the button listeners.
+        this.setSeverityButtonListeners(trackingItem);
+        this.setTimeButtonListener(trackingItem);
+        this.setDurationButtonListener(trackingItem);
+        this.setPreMedsButtonListener(trackingItem);
+        this.setPostMedsButtonListener(trackingItem);
+        this.setAddNoteButtonListener(trackingItem);
+    }
+
+    private void setLogObservers(@NonNull TrackingItem trackingItem) {
+        final LiveData<SymptomLog> logLiveData = Transformations.map(this.viewModel.getLoggedElements(), elements -> {
+            for (SymptomLog log : elements) {
+                if (log.getTrackingItem().getIdentifier().equals(trackingItem.getIdentifier())) {
+                    return log;
+                }
+            }
+
+            return null;
+        });
+
+        final LiveData<Integer> severityLiveData = Transformations
```

Comment:
It seems you could just subscribe all these to logLiveData and have them pick out the field they want. They all need to nullCheck anyway.

Reply:
So I think it is better to this since with subscribing to the logLiveData I would have to store the previous state and ensure that the new state is different and that the UI needs to be changed. By doing a transformation if the logLiveData changes, but the fields value doesn't the field's live data never notifies the observer.